ui-particles-angular/projects/ui-policy-studio-angular/src/lib/components/flow-form-dialog/gio-ps-flow-form-dialog.component.spec.ts

```diff
@@ -0,0 +1,230 @@
+import { Component } from '@angular/core';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { HarnessLoader } from '@angular/cdk/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { MatButtonHarness } from '@angular/material/button/testing';
+import { InteractivityChecker } from '@angular/cdk/a11y';
+import { MatIconTestingModule } from '@angular/material/icon/testing';
+
+import { GioPolicyStudioModule } from '../../gio-policy-studio.module';
+import { fakeChannelFlow } from '../../models/index-testing';
+import { FlowVM } from '../../gio-policy-studio.model';
+
+import {
+  GioPolicyStudioFlowFormDialogComponent,
+  GioPolicyStudioFlowFormDialogData,
+  GioPolicyStudioFlowFormDialogResult,
+} from './gio-ps-flow-form-dialog.component';
+import { GioPolicyStudioFlowFormDialogHarness } from './gio-ps-flow-form-dialog.harness';
+
+@Component({
+  selector: 'gio-dialog-test',
+  template: `<button mat-button id="open-dialog" (click)="openDialog()">Open dialog</button>`,
+})
+class TestComponent {
+  public flow?: FlowVM;
+  public flowToEdit?: FlowVM;
+  public entrypoints = ['entrypoint1', 'entrypoint2'];
+  constructor(private readonly matDialog: MatDialog) {}
+
+  public openDialog() {
+    this.matDialog
+      .open<GioPolicyStudioFlowFormDialogComponent, GioPolicyStudioFlowFormDialogData, GioPolicyStudioFlowFormDialogResult>(
+        GioPolicyStudioFlowFormDialogComponent,
+        {
+          data: {
+            entrypoints: this.entrypoints,
+            flow: this.flowToEdit,
+          },
+          role: 'alertdialog',
+          id: 'testDialog',
+        },
+      )
+      .afterClosed()
+      .subscribe(flow => {
+        if (flow) {
+          this.flow = flow;
+        }
+      });
+  }
+}
+
+describe('GioPolicyStudioFlowFormDialogComponent', () => {
+  let component: TestComponent;
+  let fixture: ComponentFixture<TestComponent>;
+  let loader: HarnessLoader;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [TestComponent],
+      imports: [GioPolicyStudioModule, MatDialogModule, NoopAnimationsModule, MatIconTestingModule],
+    }).overrideProvider(InteractivityChecker, {
+      useValue: {
+        isFocusable: () => true, // This traps focus checks and so avoid warnings when dealing with
+        isTabbable: () => true, // hidden elements
+      },
+    });
+
+    fixture = TestBed.createComponent(TestComponent);
+    component = fixture.componentInstance;
+    loader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+  });
+
+  it('should return false on cancel', async () => {
+    component.flow = {
+      _id: 'test-id',
+      ...fakeChannelFlow(),
+    };
+    await componentTestingOpenDialog();
+
+    const flowFormDialogHarness = await loader.getHarness(GioPolicyStudioFlowFormDialogHarness);
+    await flowFormDialogHarness.cancel();
```

Comment:
Maybe we should perform some changes here and check they are not saved on cancel ?